extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -429,10 +429,16 @@ public static boolean areHardcodedYoutubeMusicKeysValid() throws IOException, Re
         return youtubeMusicKeys = new String[]{key, clientName, clientVersion};
     }
 
+
+
     @Nullable
     public static String getUrlFromNavigationEndpoint(JsonObject navigationEndpoint) throws ParsingException {
         if (navigationEndpoint.has("urlEndpoint")) {
             String internUrl = navigationEndpoint.getObject("urlEndpoint").getString("url");
+            if (internUrl.startsWith("https://www.youtube.com/redirect?")) {
+                internUrl = internUrl.substring(23);
+            }
+
             if (internUrl.startsWith("/redirect?")) {
```

Comment:
```suggestion
            } else if (internUrl.startsWith("/redirect?")) {
```

Reply:
No, it's expected. When it starts with `https://youtube.com/redirect?`, I only remove the `https://youtube.com` part so that it falls in this condition.

Reply:
oh ok. please add that info as comment. I would not expect that behaviour and a comment is better to understand than counting letters.

Reply:
done